src/FSharp.Core/local.fs

```diff
@@ -14,6 +14,11 @@ module internal DetailedExceptions =
         let msg = String.Format (format, paramArray)
         raise (new ArgumentException (msg, arg))
 
+    /// takes an argument, a formatting string, a param array to splice into the formatting string
+    let inline invalidArgOutOfRangeFmt (arg:string) (format:string) paramArray =
+        let msg = String.Format (format, paramArray)
+        raise (new ArgumentOutOfRangeException (arg, msg))
```

Comment:
I think `ArgumentOutOfRangeException` also allows a structured property indicating the `index` which was passed in, which would be handy for structured processing. (i.e. not having to parse it out of the `msg`).
This would be a good addition.

Reply:
I looked into if this can be elaborated better, didn't come with a clear way, but we can add some extra data here later.